### DISCUSSION TITLE
Webman commands do not work through the web browser

### DISCUSSION THREAD

**Discussion by sandrostoregames** [aldostools/webMAN-MOD#1115] [Ideas]
Webman commands do not work through the web browser when manual proxy configuration is used, it only works on xmb, I use port 8080 with the local ip with the android proxy server app.

I would like you to do a test and find out why the browser gets this error when activating the webman command.

app used
https://github.com/Edw590/PS3-Proxy-Server-for-Android/releases

videos how the error occurs
https://drive.google.com/drive/folders/1NwwDBC7xH2wycaV6wjT40jNYbIU-6Geh?usp=drive_link

**Comment by aldostools** [comment]
WebMAN-MOD uses the port 80 to listen web server requests.

You're using a different port. Check if you can access it the ports 80 (http) or 21 (ftp).

**Reply by sandrostoregames** [reply]
The ps3 proxy android application only allows you to use port 8080, can you not allow webman to read port 8080 too?  it would help a lot

**Reply by sandrostoregames** [reply]
![image](https://github.com/user-attachments/assets/35e8c225-d363-4c64-8086-0d49132309e9)

**Reply by aldostools** [reply]
Honestly I don't understand what you're trying to do. webMAN is a server that listen and replies on port 80. The proxy is another server used typically as intermediary to connect to Internet or other servers in a LAN.

WebMAN doesn't use proxy.

Why don't you connect directly to PS3 IP through port 80?

If you can't, you should ask to the proxy developer to allow the connection.

**Reply by sandrostoregames** [reply]
the app is used to increase the transfer rate when downloading files directly from the ps3, the default PSN settings leave it with huge latency with other cloud services and ends up taking a long time to download large files such as games and emulators, I have already changed DNS but only solve with proxy server

**Reply by sandrostoregames** [reply]
For example, I use webman commands to download

http://localhost/popup.ps3/StartDownload!;/wait.ps3?3;/install.ps3?url=
http://localhost/popup.ps3/StartDownload!;/wait.ps3?3;/download.ps3?to=/dev_hdd0/PS3ISO&url=

but with port 8080 activated, the commands do not work in the browser, only directly from the XMB, if the webman could read these commands with port 8080 it would be perfect

**Reply by aldostools** [reply]
Have you configured the proxy in XMB > Settings > Network Settings? 

The PS3 will use the proxy to connect to Internet. XMB connects to webMAN MOD using local host back loop through port 80.

If you're trying to connect to webMAN MOD remotely, use the PS3 IP address on port 80. But the remote device must be in the same network segment.

**Reply by sandrostoregames** [reply]
webman works perfectly with the proxy activated, but the commands are only executed directly from a command xml, if any code is inserted into a website or html they are not executed by the browser, I wanted them to be executed, as you say it should be because of the port, it would be interesting if the webman responded to his commands with port 8080

**Reply by aldostools** [reply]
The commands install.ps3 and download.ps3 uses the network settings which you have configured with proxy. Try using the PS3 IP instead of localhost in the URL parameter. For the proxy localhost is your phone, not the PS3.

**Reply by sandrostoregames** [reply]
In this case I would use my IP, but as other people each would have a different IP, it would have to be a situation that worked for everyone just by activating the ps3 proxy server

**Reply by sandrostoregames** [reply]
Would it be possible to change localhost to something that completes the url according to each ps3 IP as if it were a keyword

**Comment by sandrostoregames** [comment]
You could put an option in your mod so that we can change the port according to any proxy we use and leave 80 as the default and change the option whenever we want, many people use this app but are unable to use it with the webman precisely because of this error

**Reply by aldostools** [reply]
Sorry but I don't have plans to make changes for the default WWW port.

You can make your own custom version changing this line in `main.c`.
```C
#define WWWPORT			(80)
```
